File: Medium/find_first_and_last_position_of_element_in_sorted_array.php
<?php

class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function searchRange($nums, $target) {
        $collectRange = array();
        
        foreach($nums as $i => $n) {
            if($n == $target) {
                array_push($collectRange, $i);
            }
        }

        if(count($collectRange) == 0) {
            return array(-1,-1);
        } else if(count($collectRange) == 1) {
            return array_pad($collectRange, 2, $collectRange[0]);
        } else {
            return array($collectRange[0], $collectRange[count($collectRange)-1]);
        }
    }
}

$solution = new Solution;

print_r($solution->searchRange(array(5,7,7,8,8,10), 8));
print_r($solution->searchRange(array(1), 1));
print_r($solution->searchRange(array(3,3,3), 3));
print_r($solution->searchRange(array(0,0,1,2,2), 2));

File: Medium/find_first_and_last_position_of_element_in_sorted_array.py
class Solution(object):
    def searchRange(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        collectRange = []
        for i, n in enumerate(nums):
            if n == target:
                collectRange.append(i)
                
        if len(collectRange) == 0:
            return [-1, -1] 
        elif len(collectRange) == 1:
            return 2*collectRange
        else:
            return [collectRange[0], collectRange[-1]]
            
print(Solution().searchRange([5,7,7,8,8,10], 8))
print(Solution().searchRange([1], 1))
print(Solution().searchRange([3,3,3], 3))
print(Solution().searchRange([0,0,1,2,2], 2))